### PR TITLE
optimize gh workflows

### PR DESCRIPTION
**Pull request by Huskydog9988**
Only run workflow for a subproject when its files have been changed

### CODE REVIEW

File: .github/workflows/build-server.yml [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
-      with:
-        node-version: 18
-    - run: cd server && yarn
-    - run: cd server && yarn build
-    - run: cd server && yarn test
+      - uses: actions/checkout@v3
+      - name: Use latest Node.js LTS
+        uses: actions/setup-node@v3
+        with:
+          node-version: "lts"
```
Comment (Protonull) [inline .github/workflows/build-server.yml:19]: Probably better to stick with a defined version

File: .github/workflows/build-server.yml [hunk 2]
```diff
@@ -1,16 +1,26 @@
 name: Build+Test Server
 
-on: [ push, pull_request ]
+on:
+  push:
+    paths:
+      - "server/**/*"
+  pull_request:
+    paths:
+      - "server/**/*"
```
Comment (Gjum) [inline .github/workflows/build-server.yml:9]: very cool, I didn't know this was possible

File: .github/workflows/build-server.yml (hunk 1 above)
Comment (Gjum) [reply .github/workflows/build-server.yml:19]: We could test with both an old LTS version (for existing installs) and the latest node version.

**Review by Gjum — DISMISSED** [review]

File: .github/workflows/build-server.yml (hunk 1 above)
Comment (Huskydog9988) [reply .github/workflows/build-server.yml:19]: How about both lts and latest?

File: .github/workflows/build-server.yml (hunk 2 above)
Comment (Huskydog9988) [reply .github/workflows/build-server.yml:9]: Yea, I just learn about it maybe a month back

**Review by Gjum — APPROVED** [review]

**Pull request merged**